tests/src/unit-class_parser.cpp

```diff
@@ -302,6 +303,36 @@ void comments_helper(const std::string& s)
     }
 }
 
+void validateFn(const std::string& original_string, const json& j, const json& check)
+{
+    CHECK(j == check);
+    CHECK(j.get_start_position() == 0);
+    CHECK(j.get_end_position() == original_string.size());
+}
+
+void validate_start_end_pos_for_nested_obj(std::string& nested_type_json_str, const std::string& root_type_json_str, const json& expected_json, json::parser_callback_t cb = nullptr)
+{
+    json j;
+
+    // 1. If callback is provided, use callback version of parse()
+    if (cb)
+    {
+        j = json::parse(root_type_json_str, cb);
+    }
+    else
+    {
+        j = json::parse(root_type_json_str);
+    }
+
+    // 2. Check if the generated JSON is as expected
+    validateFn(root_type_json_str, j, expected_json);
```

Comment:
Dropping a comment here regarding the assumptions this is making and the lack of coverage for leading/trailing characters